setup.py

```diff
@@ -200,7 +200,7 @@ def run(self):
             "scipy",  # Snowpark pandas 3rd party library testing
             "statsmodels",  # Snowpark pandas 3rd party library testing
             "scikit-learn==1.5.2",  # Snowpark pandas scikit-learn tests
-            "plotly",  # Snowpark pandas 3rd party library testing
+            "plotly<6.0.0",  # Snowpark pandas 3rd party library testing
```

Comment:
Could we add a comment saying that we're capping plotly because plotly 6 requires from_dict() instead of the interchange protocol?

Reply:
I added one now. This for our comment in setup and not the interop doc right?

Reply:
This is the PR: https://github.com/snowflakedb/snowpark-python/pull/2759

Reply:
I suggest that we merge #2759 first, but @sfc-gh-lmukhopadhyay will need to update the query counts in these tests.

Reply:
Never mind! I forgot that plotly 6 has not been released yet. So we should keep this cap and remove it after plotly 6 comes out: https://pypi.org/project/plotly/#history

Reply:
Even with support for `from_dict` the performance would be quite slow due to the large # of queries issued right? If this is not a blocker for us, I'll remove the cap

Reply:
We don't need the cap if the performance due to the large # of queries issued aren't a blocker. And if we're planning on removing it after the v6.0.0. release anyway, we don't need to add it now right?

Reply:
Our comments may have had a race condition, but I think my comment explaining why we do need the cap still stands:

> Never mind! I forgot that plotly 6 has not been released yet. So we should keep this cap and remove it after plotly 6 comes out: [pypi.org/project/plotly#history](https://pypi.org/project/plotly/#history)

Reply:
Got it, so we can have the cap now, and once to_dict is supported (and the 6.0.0 release is out), we can update the query counts and remove the cap

Reply:
PS @sfc-gh-mvashishtha I don't think this comment is necessary anymore if we're merging from_dict before this PR

Reply:
Ok, let's instead say that we need to update the query counts and confirm that plotly 6+ will work once it comes out.